internal/auth/oidc/repository_account.go

```diff
@@ -12,6 +12,84 @@ import (
 	"github.com/hashicorp/boundary/internal/oplog"
 )
 
+// CreateAccount inserts a into the repository and returns a new Account
+// containing the account's PublicId. a is not changed. a must contain a
+// valid AuthMethodId. a must not contain a PublicId. The PublicId is
+// generated and assigned by this method. a must not contain an IssuerId.
+// The IssuerId is retrieved from the auth method. If it does not contain an
+// IssuerId an error is returned.
+//
+// a must contain a valid SubjectId. a.SubjectId must be unique for an
+// a.AuthMethod/Issuer pair.
+//
+// Both a.Name and a.Description are optional. If a.Name is set, it must be
+// unique within a.AuthMethodId.
+func (r *Repository) CreateAccount(ctx context.Context, scopeId string, a *Account, opt ...Option) (*Account, error) {
+	const op = "oidc.(Repository).CreateAccount"
+	if a == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "missing Account")
+	}
+	if a.Account == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "missing embedded Account")
+	}
+	if a.AuthMethodId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "missing auth method id")
+	}
+	if a.SubjectId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "missing subject id")
+	}
+	if a.PublicId != "" {
+		return nil, errors.New(errors.InvalidParameter, op, "public id must be empty")
+	}
+	if a.IssuerId != "" {
+		return nil, errors.New(errors.InvalidParameter, op, "issuer id must be empty")
+	}
+	if scopeId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "missing scope id")
+	}
+
+	a = a.Clone()
+
+	am, err := r.LookupAuthMethod(ctx, a.AuthMethodId)
```

Comment:
hum... I'm sort of rethinking this.  If we only allow operators to provision accounts for the current auth_oidc_method.issuer then how will operators provision accounts if they need to prepare for an upcoming issuer change?  I'm not sure how real this use-case is but it's yet again an example of Boundary forcing a specific workflow on the operator which may be incorrect.

Reply:
I'm fine with having the current auth method's issuer be the default, but I"m not sure we should force that.

Reply:
If we do decide to move forward with either providing a default issuer or enforcing this constraint on operators, then perhaps a comment about how looking up this now (and the race condition is creates) is acceptable?

Reply:
I updated the PR to reflect this.  It is now optional to provide the issuer when creating an account.